examples/ghga_auth/ghga_auth/inject.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Dependency injection logic"""
+
+from collections.abc import AsyncGenerator
+from contextlib import asynccontextmanager
+from typing import Optional
+
+from ghga_auth import dummies
+from ghga_auth.config import Config
+from ghga_auth.router_config import get_configured_app
+from ghga_service_commons.auth.context import AuthContextProtocol
+from ghga_service_commons.auth.ghga import AuthContext, GHGAAuthContextProvider
+from ghga_service_commons.utils.context import asyncnullcontext
+
+
+@asynccontextmanager
+async def prepare_auth_provider(
```

Comment:
Same as above

Reply:
See [3c7102b](https://github.com/ghga-de/ghga-service-commons/pull/30/commits/3c7102b4e49180728c6a9c3bfff3f60b65137d93), hopefully that's what you meant.